### PR TITLE
New Coq plugin for general proof automation

### PR DESCRIPTION
**Pull request by ckeller**
This PR introduces an opam package for `sniper`, a new Coq plugin for general proof automation extending the `SMTCoq` plugin

### CODE REVIEW

File: extra-dev/packages/coq-sniper/coq-sniper.dev+8.13/opam [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+build: [
+  [make "-j%{jobs}%"]
+]
+install: [
+  [make "install"]
+]
+depends: [
+  "ocaml" {>= "4.07.1" }
+  "coq" {>= "8.13~" & < "8.14~"}
+  "coq-metacoq" {= "1.0~beta2+8.13"}
```
Comment (silene) [inline extra-dev/packages/coq-sniper/coq-sniper.dev+8.13/opam:21]: Is there a reason to force such a very specific version of MetaCoq? Are you expecting your package to break with the actual 1.0 release of MetaCoq?

**Review by silene — CHANGES_REQUESTED** [review]

File: extra-dev/packages/coq-sniper/coq-sniper.dev+8.11+8.12/opam [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  "ocaml" {>= "4.07.1" }
+  "coq" {>= "8.11~" & < "8.13~"}
+  "coq-metacoq"
+  "coq-smtcoq"
+]
+tags: [ 
+  "category:Computer Science/Decision Procedures and Certified Algorithms/Decision procedures"
+  "category:Miscellaneous/Coq Extensions"
+  "keyword: SMT"
+  "keyword: automation"
```
Comment (silene) [inline extra-dev/packages/coq-sniper/coq-sniper.dev+8.11+8.12/opam:28]: It seems some .vo files get installed, so there should be a tag `"logpath:Sniper"`.

**Comment by ckeller** [comment]
Hi @silene 
I applied all the changes (and also corrected a (very bad) typo) except about the 1.0 release of Metacoq: I do not see this release? In opam, all the versions have ~alpha or ~beta.

**Comment by silene** [comment]
What I meant is that your `dev` package will be stuck forever with MetaCoq `1.0~beta2`. So, unless you are already sure that Sniper will be incompatible with the actual 1.0 release of MetaCoq, there is absolutely no reason to put such a harsh constraint. It should instead be `>= "1.0~beta2"`, `>= "1.0~beta"`, or just nothing, depending on how incompatible Sniper is with the various prereleases of MetaCoq.

**Comment by ckeller** [comment]
I thought it was safer to update the `opam` file after checking that everything is fine, but ok, I have removed the required version.

**Comment by silene** [comment]
One job failed due to dependencies (I expected this one), but the three other ones timed out. This is a bit worrying. Is it really supposed to take more than three hours to compile Sniper and its dependencies or is there an issue with the Opam package?

**Comment by silene** [comment]
Related (and perhaps naive) question: Do you need the full-blown MetaCoq or is a subset of it (e.g., `meta-coq-template`) sufficient for Sniper?

**Comment by ckeller** [comment]
On my computer, it takes approximately 20 minutes to build everything. The longer is indeed MetaCoq, we can restrict to three packages for the moment if needed.

**Comment by silene** [comment]
I have merged the pull request in the meantime. But it would be great to reduce the dependencies to those strictly needed, in a followup patch. By the way, the dependency on OCaml should also be removed, since Sniper does not provide any OCaml file, as far as I can tell.

**Pull request merged**